include/sortcheck.h

```diff
@@ -239,8 +255,14 @@ inline bool binary_search_checked(_ForwardIterator __first,
                                   _ForwardIterator __last,
                                   const _Tp &__val,
                                   const char *file, int line) {
-  Compare<_Tp> compare;
+#ifdef SORTCHECK_SUPPORT_COMPARELESS_API
+  Compare2<SORTCHECK_DECLTYPE(*__first), _Tp> compare;
   return binary_search_checked(__first, __last, __val, compare, file, line);
+#else
+  file = file;
+  line = line;
```

Comment:
Is it used to suppress the unused variable warning?
If so, I suggest to introduce such macro:
```cpp
#define SORTCHECK_UNUSED(x) ((void)(x))
```
And then:
```suggestion
  SORTCHECK_UNUSED(file);
  SORTCHECK_UNUSED(line);
```